functions/api/action.js

```diff
@@ -309,28 +327,42 @@ router.post('/rejectRequest', async (req, res, next) => {
         return;
     }
     const { toBranch, fromBranch, fromBranchName, productName, pendingRequestsId } = req.body
-
+    const toBranchRef = db.collection(constants.BRANCHES).doc(toBranch)
+    const fromBranchRef = db.collection(constants.BRANCHES).doc(fromBranch)
     //check if pending requests exist
-    const toBranchPendingReqRef = db.collection(constants.BRANCHES).doc(toBranch).collection(constants.PENDING_REQUESTS).doc(pendingRequestsId)
-    const fromBranchPendingReqRef = db.collection(constants.BRANCHES).doc(fromBranch).collection(constants.PENDING_REQUESTS).doc(pendingRequestsId)
+    const toBranchPendingReqRef = toBranchRef.collection(constants.PENDING_REQUESTS).doc(pendingRequestsId)
+    const fromBranchPendingReqRef = fromBranchRef.collection(constants.PENDING_REQUESTS).doc(pendingRequestsId)
     const pendingReqDocs = await db.getAll(toBranchPendingReqRef, fromBranchPendingReqRef)
     const toBranchPendingReqDoc = pendingReqDocs[0]
     const fromBranchPendingReqDoc = pendingReqDocs[1]
-
     if (!toBranchPendingReqDoc.exists || !fromBranchPendingReqDoc.exists) {
         const error = new Error('No pending requests to reject')
         error.statusCode = 404
         next(error)
         return
     }
+    const toBranchData = {
+        ...toBranchPendingReqDoc.data(),
+        id: toBranchPendingReqDoc.id,
+        accepted: false
+    }
+    const fromBranchData = {
+        ...fromBranchPendingReqDoc.data(),
+        id: fromBranchPendingReqDoc.id,
+        accepted: false
+    }
+    //fire and forget request state tracker
+    toBranchRef.collection("requestsHistory").add(toBranchData)
+    fromBranchRef.collection("requestsHistory").add(fromBranchData)
```

Comment:
Is this the new collection for maintaining whether the pending request was accepted or rejected ?

Reply:
One more thing, are we going to get only the rejected requests stored here ? And how about the accepted requests @neerk28 ?

Reply:
@ramyarv Both are saved in this collection. There is a field accepted (true/false).

Reply:
sure @shrikantdange1208  . just curious to know, we could have added the status of request in the pending requests itself right like PENDING/ACCEPTED/REJECTED. why we need a separate collection for it ?

Reply:
we could do that as well, its just a design choice. in get of pending requests, UI should pass a filter for state: PENDING to show in inventory screen, and for dashboard they could pass state:ACCEPTED/REJECTED.